Assets/Scripts/Models/HealthSystem.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+
+[Serializable]
+public class HealthSystem
+{
+    //all of this data should probably be generated at runtime or loaded from a file
+    public float Value
+    {
+        get
+        {
+            return _currentHealth;
+        }
+
+        private set
+        {
+            if (Invincible == true) return;
+            //We have been Revived!
+            if (_currentHealth <= 0 && value > _currentHealth)
```

Comment:
Yes, for reviving, if the characters health is at  0 or less then he is dead. But if the current value were adding to his health is greater then 0 then he is being revived.